### PR TITLE
Bump CDI API to v1.50.0

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ShellyKa13** [comment]
not sure about it but should this stuff happen through kubevirt-ci? for example I see this PR: https://github.com/kubevirt/kubevirtci/pull/793
I remember that I wanted to bump CDI and i've been told that there is a script that should do it automatically. I talked with @dhiller at the time

**Comment by mhenriks** [comment]
Should update here as well:  https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/api/go.mod#L11

**Comment by akalenyu** [comment]
> not sure about it but shouldn't this stuff happen through kubevirt-ci? for example I see this PR: [kubevirt/kubevirtci#793](https://github.com/kubevirt/kubevirtci/pull/793) I remember that I wanted to bump CDI and i've been told that there is a script that should do it automatically. I talked with @dhiller at the time

This one takes care of bumping the CDI containers that run in the cluster, I believe Arnon is building up to use some new APIs in kubevirt controllers

**Comment by dhiller** [comment]
> not sure about it but shouldn't this stuff happen through kubevirt-ci? for example I see this PR: [kubevirt/kubevirtci#793](https://github.com/kubevirt/kubevirtci/pull/793) I remember that I wanted to bump CDI and i've been told that there is a script that should do it automatically. I talked with @dhiller at the time

CDI inside KubeVirtCI is updated automatically. We don't have automation to update the CDI api inside KubeVirt. Can that even be automated? Or do I not understand the question here?

**Comment by dhiller** [comment]
> pull-kubevirt-fossa	[0e7fb0c](https://github.com/kubevirt/kubevirt/commit/0e7fb0c354bdadfc6cc59da4106be6788eaee8ab)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7906/pull-kubevirt-fossa/1536615858141401088)	false	`/test pull-kubevirt-fossa`

Looks like FOSSA thinks it found a license issue in the PR regarding CDI: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/7906/pull-kubevirt-fossa/1536615858141401088#1:build-log.txt%3A160

https://app.fossa.com/projects/custom+13072%2Fgit@github.com:kubevirt%2Fkubevirt.git/refs/branch/main/96c6d3ccd73b8c7e6ab7213dbfad6d942467fbc2

I'll take a look at the issues now.

**Comment by dhiller** [comment]
IIUC FOSSA is complaining about some GPL'ed stuff that CDI itself uses. So I think it's fair to say that KubeVirt is good, as it's only using the CDI API.

BUT I am not sure whether there is something to fix at CDI v1.50.0 regarding those GPL'ed libraries:
* https://github.com/kubevirt/containerized-data-importer/tree/v1.50.0/vendor/libguestfs.org/libnbd#license
* https://github.com/kubevirt/containerized-data-importer/blob/v1.50.0/vendor/github.com/gorhill/cronexpr/GPLv3

@arnongilboa should I open an issue in CDI repo for this?

**Comment by arnongilboa** [comment]
> IIUC FOSSA is complaining about some GPL'ed stuff that CDI itself uses. So I think it's fair to say that KubeVirt is good, as it's only using the CDI API.
> 
> BUT I am not sure whether there is something to fix at CDI v1.50.0 regarding those GPL'ed libraries:
> 
> * https://github.com/kubevirt/containerized-data-importer/tree/v1.50.0/vendor/libguestfs.org/libnbd#license
> * https://github.com/kubevirt/containerized-data-importer/blob/v1.50.0/vendor/github.com/gorhill/cronexpr/GPLv3
> 
> @arnongilboa should I open an issue in CDI repo for this?

Yep, good idea.

File: go.mod [hunk 1]
```diff
@@ -74,8 +74,8 @@ require (
 	kubevirt.io/api v0.0.0-00010101000000-000000000000
 	kubevirt.io/client-go v0.0.0-00010101000000-000000000000
 	kubevirt.io/containerized-data-importer v1.41.0
```
Comment (brybacki) [inline go.mod:76]: here we see version 1.41 and below on line 198 replace to 1.42? why

**Pull request merged**